core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java

```diff
@@ -734,9 +734,60 @@ public Result visit(Aggregate e) {
     RelTrait relTrait = e.getTraitSet().getTrait(PivotRelTraitDef.instance);
     if (relTrait != null && relTrait instanceof PivotRelTrait) {
       if (((PivotRelTrait) relTrait).isPivotRel()) {
+        List<SqlNode> selectList = builder.select.getSelectList();
```

Comment:
code in this class needs refactoring , can be drawn into diff methods, with proper documeting of the intend of each condition

Reply:
yes , i have refactored the code and used different methods for different behaviors.